Samples/Excel.OfflineStorageAddin/README.md

```diff
@@ -0,0 +1,112 @@
+
+**Note**: Depending on the type and size of data you wish to offline, you may wish to look into other offline storage options. If you'd like offline capabilities to persist in the file, you may wish to store your add-in's data in [Office Settings](https://docs.microsoft.com/en-us/javascript/api/office/office.settings?view=office-js). Local storage is also limited in that it can only cache up to 5 MB of information. To store larger amounts of data offline, you may wish to consider using [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API); however, as of now, IndexedDB is not supported by all browsers used by Office add-ins, and may cause your add-in to fail on some computers.
+
+## Build and run the sample
+
+To get this sample, clone or download this repository. In the command line, navigate to the **Excel.OfflineStorageAddin** folder from your root directory.
+```
+$ git clone https://github.com/OfficeDev/PnP-OfficeAddins.git
+$ cd PnP-OfficeAddins
+$ cd Samples
+$ cd Excel.OfflineStorageAddin
+```
+You can try out this sample by running the following commands in the command line:
+```
+# this will download the node modules needed to run this add-in
+$ npm install
+
+# this will build the add-in 
+$ npm run build
+
+# this will start the server to run your add-in on Excel on your desktop
+$ npm run start
+
+# this will start the server to run your add-in on Excel on the web
+$ npm run start:web
+```
+## Key parts of this sample
+
+Navigate to ``Excel.OfflineStorageAddin/src/taskpane/taskpane.js`` to find the implementation of local storage described below. 
+
+### Implementing local storage to offline data
+The ``Excel.OfflineStorageAddin/src/taskpane/taskpane.js`` file contains the `loadTable()` function, which makes use of local storage to display a table of basketball player stats when a user loses connection.
+
+In the sample code, the `loadTable()` function first checks if the basketball player data was previously cached into local storage, as shown in the code below. If it exists, the data is parsed from JSON into a readable text format before being passed to `createTable()`, a function which creates a table from the given data. 
+
+```
+        if (localStorage.DraftPlayerData) {
+            var dataObject = JSON.parse(localStorage.DraftPlayerData);
+            createTable(dataObject);
+        }
+```
+
+If the data was not previously cached, `loadTable()` attempts to access ``sampleData.js``, which is the file containing the data, through an AJAX call. If this attempt is successful, the function passes the data returned from the file to `createTable()` to produce a table. The data is also converted into a JSON object, which is cached into local storage. However, if the function is unable to access the ``sampleData.js`` file, the function returns an error to the console, notifying the add-in. This process is shown in the following code:

```

Comment:
```suggestion
If the data wasn't previously cached, `loadTable()` attempts to access the offline data file, *sampleData.js*, through an AJAX call. If this attempt is successful, the function passes the data returned from the file to `createTable()` to produce a table. The data is also converted into a JSON object, which is cached into local storage. However, if the function is unable to access the *sampleData.js* file, the function returns an error to the console. This process is shown in the following code:
```
How will an end user know if an error has occurred? Do you have an error that shows up in the taskpane or something?